acceptance/tests/puppet_device_test.rb

```diff
@@ -1,74 +1,73 @@
 test_name "puppet device is able to run and configure a node"
 
+tag 'server'
+
 teardown do
-  on(master, "[ -f /etc/puppetlabs/puppet/puppet.conf.bak ] && mv /etc/puppetlabs/puppet/puppet.conf.bak /etc/puppetlabs/puppet/puppet.conf")
   # revert permission changes to reset state for other tests
   on(master, "puppet agent -t --server #{master}")
 end
 
-# set sensible defaults
-on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
-on(master, "puppet config set server $(hostname --fqdn)")
-
-codedir = "#{master.puppet['codedir']}/modules"
-confdir = master.puppet['confdir']
-common_options="--config /tmp/puppet.conf --debug --trace -w0"
-
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
-
-apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
-# create and configure test user to trigger PUP-9642
-user {
-  'test':
-    ensure => 'present',
-}
-
-# testfile for --apply
-file {
-  '/tmp/spinner.pp':
-    content => "spinner { '999': ensure => 'absent' }"
-}
-
-# install and configure the Resource API
-service { 'puppetserver': }
-include 'resource_api::server', 'resource_api::agent'
+with_puppet_running_on(master, {}) do
+  on(master, "puppet config set server $(hostname --fqdn)")
+
+  codedir = "#{master.puppet['codedir']}/modules"
+  confdir = master.puppet['confdir']
+  common_options="--debug --trace -w0"
+
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
+
+  apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
+  # create and configure test user to trigger PUP-9642
+  user {
+    'test':
+      ensure => 'present',
+  }
+
+  # testfile for --apply
+  file {
+    '/tmp/spinner.pp':
+      content => "spinner { '999': ensure => 'absent' }"
+  }
+
+  # install and configure the Resource API
+  include 'resource_api::server', 'resource_api::agent'
 MANIFEST
 
-on(master, "cp -v #{confdir}/puppet.conf /tmp/puppet.conf")
```

Comment:
why no more copying first?

Reply:
Because `with_puppet_running_on` already backs up the config file. I didn't think it was necessary to use a temp one in addition. So I updated this to just alter and use the default config, and it gets put back to the original state at the end automatically.

I tried running this with a test before that didn't clean up after itself all the way, and a test after that would be unhappy with the new user, and they all passed.